compiler/luci/pass/src/FuseSliceWithTConvPass.cpp

```
@@ -0,0 +1,289 @@
+/*
+ * Copyright (c) 2023 Samsung Electronics Co., Ltd. All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "luci/Pass/FuseSliceWithTConvPass.h"
+
+#include <luci/IR/CircleNodes.h>
+#include <luci/Profile/CircleNodeOrigin.h>
+#include <luci/Service/CircleNodeClone.h>
+#include <luci/Service/Nodes/CircleConst.h>
+
+namespace
+{
+
+#define RETURN_FALSE_UNLESS(cond) \
+  if (not(cond))                  \
+    return false;
+
+inline int32_t compute_input_size(luci::Padding padding, int32_t image_size, int32_t filter_size,
+                                  int32_t stride)
+{
+  switch (padding)
+  {
+    case luci::Padding::SAME:
+      return (image_size + stride - 1) / stride;
+    case luci::Padding::VALID:
+      return (image_size + stride - filter_size) / stride;
+    default:
+      assert(false);
+      return 0;
```

Please throw an exception in this case.

---

Ahh. ok.

---

done